cmd/cogito/main_test.go

```diff
@@ -62,14 +70,45 @@ func TestRunPutSmokeIntegrationSuccess(t *testing.T) {
 	var out bytes.Buffer
 	var logOut bytes.Buffer
 	inputDir := testhelp.MakeGitRepoFromTestdata(t, "../../cogito/testdata/one-repo/a-repo",
-		testhelp.HttpsRemote("the-owner", "the-repo"), "dummySHA", "dummyHead")
+		testhelp.HttpsRemote("the-owner", "the-repo"), "dummySHA", wantGitRef)
+	t.Setenv("COGITO_GITHUB_API", ts.URL)
 
 	err := run(in, &out, &logOut, []string{"out", inputDir})
 
 	assert.NilError(t, err, "\nout: %s\nlogOut: %s", out.String(), logOut.String())
+	ts.Close() // Avoid races before following asserts.
```

Comment:
As the comment says, this is a "belt and suspenders" safety. Already the code in the test server is written in a way to ensure serialization. Close() is blocking and returns when the test server is shut down.